dev/PushNotifications/PushNotificationChannel.cpp

```diff
@@ -6,21 +6,55 @@
 
 namespace winrt::Microsoft::Windows::PushNotifications::implementation
 {
+    winrt::Windows::Foundation::TypedEventHandler<
+        Microsoft::Windows::PushNotifications::PushNotificationChannel,
+        Microsoft::Windows::PushNotifications::PushNotificationReceivedEventArgs> s_typedEventHandler;
+
     PushNotificationChannel::PushNotificationChannel(winrt::Windows::Networking::PushNotifications::PushNotificationChannel const& channel)
     {
         m_channel = channel;
     }
     winrt::Windows::Foundation::Uri PushNotificationChannel::Uri()
     {
+        auto lock = m_lock.lock();
         winrt::Windows::Foundation::Uri pushChannelUri{ m_channel.Uri() };
         return pushChannelUri;
     }
     winrt::Windows::Foundation::DateTime PushNotificationChannel::ExpirationTime()
     {
+        auto lock = m_lock.lock();
         return m_channel.ExpirationTime();
     }
     void PushNotificationChannel::Close()
     {
+        auto lock = m_lock.lock();
         m_channel.Close();
     }
+
+    winrt::event_token PushNotificationChannel::PushReceived(winrt::Windows::Foundation::TypedEventHandler<Microsoft::Windows::PushNotifications::PushNotificationChannel, Microsoft::Windows::PushNotifications::PushNotificationReceivedEventArgs> const& handler)
+    {
+        winrt::Windows::Foundation::TypedEventHandler<
+            winrt::Windows::Networking::PushNotifications::PushNotificationChannel,
+            winrt::Windows::Networking::PushNotifications::PushNotificationReceivedEventArgs> typedEventHandler
+            ([&](
+                winrt::Windows::Networking::PushNotifications::PushNotificationChannel channel,
+                winrt::Windows::Networking::PushNotifications::PushNotificationReceivedEventArgs args)
+                {
+                    Microsoft::Windows::PushNotifications::PushNotificationReceivedEventArgs reunionPushArgs =
+                        Microsoft::Windows::PushNotifications::PushNotificationReceivedEventArgs::CreateFromPushNotificationReceivedEventArgs(args);
+
+                    s_typedEventHandler(*this, reunionPushArgs);
+                });
+
+        auto lock = m_lock.lock();
+        s_typedEventHandler = handler;
```

Comment:
Why is this static and not a member variable of PushNotificationChannel?

Reply:
This should be labelled m_typedEventHandler.